### PR TITLE
AfterShip (Independent Publisher)

### PR DESCRIPTION
**Pull request by taiki-yoshida**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

- [X] I attest that the connector works and I verified by deploying and testing all the operations. 
- [X] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [X] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [X] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [X] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [X] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [X] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
image

- [X] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
image

- [X] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md. 
- Does not use OAuth

### CODE REVIEW

File: independent-publisher-connectors/AfterShip/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,2690 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "1.0.0",
+    "title": "AfterShip",
+    "description": "AfterShip - Shipment tracking API"
```
Comment (sriyen-msft) [inline independent-publisher-connectors/AfterShip/apiDefinition.swagger.json:6]: @taiki-yoshida Please provide a valid description for your connector which can be easily understandable by everyone and it should describe concisely the main purpose and value offered by your connector.

Docs - [Write a description for your connector](https://docs.microsoft.com/connectors/custom-connectors/certification-submission#write-a-description-for-your-connector)

File: independent-publisher-connectors/AfterShip/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,2690 @@
...
+  },
```
Comment (sriyen-msft) [inline independent-publisher-connectors/AfterShip/apiDefinition.swagger.json:7]: @taiki-yoshida The metadata is missing for your connector. Please make sure that you are following all the instructions mentioned in the [documentation](https://docs.microsoft.com/connectors/custom-connectors/certification-submission#step-3-add-metadata).

Docs - [Add metadata](https://docs.microsoft.com/connectors/custom-connectors/certification-submission#step-3-add-metadata)

File: independent-publisher-connectors/AfterShip/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,2690 @@
...
+                            "type": "string",
+                            "description": "phone"
+                          },
+                          "other_name": {
+                            "type": "string",
+                            "description": "other_name"
+                          },
+                          "web_url": {
+                            "type": "string",
+                            "description": "web_url"
```
Comment (sriyen-msft) [inline independent-publisher-connectors/AfterShip/apiDefinition.swagger.json:82]: @taiki-yoshida All the description values of properties are expected to be free from grammatical and spell errors. Please provide valid description values.
Docs - [Define operation and parameter summaries and descriptions](https://docs.microsoft.com/connectors/custom-connectors/certification-submission#define-operation-and-parameter-summaries-and-descriptions)

File: independent-publisher-connectors/AfterShip/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,2690 @@
...
+          {
+            "name": "body",
+            "in": "body",
+            "schema": {
+              "type": "object",
+              "properties": {
+                "reason": {
+                  "type": "string",
+                  "description": "reason",
+                  "title": "",
```
Comment (sriyen-msft) [inline independent-publisher-connectors/AfterShip/apiDefinition.swagger.json:2178]: @taiki-yoshida Empty property values can be removed until unless you have a specific reason to keep them.

File: independent-publisher-connectors/AfterShip/apiProperties.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+            "tabIndex": 2,
+            "clearText": false,
+            "required": "true"
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#f9a31a",
+    "capabilities": [],
+    "publisher": "Taiki Yoshida"
```
Comment (sriyen-msft) [inline independent-publisher-connectors/AfterShip/apiProperties.json:20]: @taiki-yoshida The `"stackOwner" property is missing. Please add it.

Docs - [publisher and stackOwner properties](https://docs.microsoft.com/connectors/custom-connectors/certification-submission#step-3a-publisher-and-stackowner-properties)

File: independent-publisher-connectors/AfterShip/apiProperties.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+          "tooltip": "Provide your API Key",
+          "constraints": {
+            "tabIndex": 2,
+            "clearText": false,
+            "required": "true"
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#f9a31a",
```
Comment (sriyen-msft) [inline independent-publisher-connectors/AfterShip/apiProperties.json:18]: @taiki-yoshida Please refer to the [template](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/apiProperties.json) we have provided for `apiProperties.json` file and default iconBrandColor for all the independent-publisher connectors is `"#da3b01"`.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

**Comment by taiki-yoshida** [comment]
@sriyen-msft all points have been resolved. Kindly review thanks!

**Comment by taiki-yoshida** [comment]
@sriyen-msft all points have been resolved. Kindly revivew thanks!

File: independent-publisher-connectors/AfterShip/apiDefinition.swagger.json (hunk 2 above)
Comment (sriyen-msft) [reply independent-publisher-connectors/AfterShip/apiDefinition.swagger.json:12]: @taiki-yoshida Please resolve this comment to get your PR reviewed again.

**Comment by natalie-pienkowska** [comment]
Thanks, Taiki for your submission and for addressing Srikanth's feedback! Looking forward to seeing your connector on the platform :)

File: independent-publisher-connectors/AfterShip/apiDefinition.swagger.json (hunk 1 above)
Comment (sriyen-msft) [reply independent-publisher-connectors/AfterShip/apiDefinition.swagger.json]: @taiki-yoshida The description of your connector still shows the same. Please do not resolve the comment until you push the changes to your PR.

File: independent-publisher-connectors/AfterShip/apiDefinition.swagger.json (hunk 2 above)
Comment (sriyen-msft) [reply independent-publisher-connectors/AfterShip/apiDefinition.swagger.json:12]: @taiki-yoshida I can not see your changes. Can you please let me know where did you add the required metadata ?

File: independent-publisher-connectors/AfterShip/apiDefinition.swagger.json (hunk 3 above)
Comment (sriyen-msft) [reply independent-publisher-connectors/AfterShip/apiDefinition.swagger.json]: @taiki-yashida I can not see changes to the description values.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]
@taiki-yoshida I have reopened all the comments as there is no change pushed to the PR.

**Comment by taiki-yoshida** [comment]
@sriyen-msft You are looking at the original commits. The updated commits have been made to the PR since then.
https://github.com/microsoft/PowerPlatformConnectors/pull/1260/commits/4f267385f1c0f1e5da8f7c6cfd74828cd49349bc

**Comment by sriyen-msft** [comment]
> @sriyen-msft You are looking at the original commits. The updated commits have been made to the PR since then. [4f26738](https://github.com/microsoft/PowerPlatformConnectors/commit/4f267385f1c0f1e5da8f7c6cfd74828cd49349bc)

@taiki-yoshida  Update - We are preparing your connector for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes between 7-10 business days. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps. Thanks

**Comment by sriyen-msft** [comment]
[[certify-connector]]

**Comment by sriyen-msft** [comment]
@taiki-yoshida We have found below swagger validation errors during prepping your connector for production deployment. There might be some error in the validation tool in the opensource repo which could not be able to catch these. Can you please fix the errors and push the latest commit to your PR.

`##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}/get/parameters/0' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-summary' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}/get/parameters/1' Value: 'x-ms-summary'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}/get/parameters/1' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-summary' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}/put/parameters/0' Value: 'x-ms-summary'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}/put/parameters/0' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-summary' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}/put/parameters/1' Value: 'x-ms-summary'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}/put/parameters/1' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-summary' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}/delete/parameters/0' Value: 'x-ms-summary'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}/delete/parameters/0' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-summary' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}/delete/parameters/1' Value: 'x-ms-summary'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}/delete/parameters/1' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-summary' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}~1retrack/post/parameters/0' Value: 'x-ms-summary'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}~1retrack/post/parameters/0' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-summary' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}~1retrack/post/parameters/1' Value: 'x-ms-summary'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}~1retrack/post/parameters/1' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-summary' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}~1mark-as-completed/post/parameters/0' Value: 'x-ms-summary'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}~1mark-as-completed/post/parameters/0' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-summary' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}~1mark-as-completed/post/parameters/1' Value: 'x-ms-summary'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1trackings~1{slug}~1{tracking_number}~1mark-as-completed/post/parameters/1' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-summary' property is required. ValuePath: 'paths/~1notifications~1{slug}~1{tracking_number}/get/parameters/0' Value: 'x-ms-summary'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1notifications~1{slug}~1{tracking_number}/get/parameters/0' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-summary' property is required. ValuePath: 'paths/~1notifications~1{slug}~1{tracking_number}/get/parameters/1' Value: 'x-ms-summary'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1notifications~1{slug}~1{tracking_number}/get/parameters/1' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-summary' property is required. ValuePath: 'paths/~1notifications~1{slug}~1{tracking_number}~1add/post/parameters/0' Value: 'x-ms-summary'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1notifications~1{slug}~1{tracking_number}~1add/post/parameters/0' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-summary' property is required. ValuePath: 'paths/~1notifications~1{slug}~1{tracking_number}~1add/post/parameters/1' Value: 'x-ms-summary'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1notifications~1{slug}~1{tracking_number}~1add/post/parameters/1' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-summary' property is required. ValuePath: 'paths/~1notifications~1{slug}~1{tracking_number}~1remove/post/parameters/0' Value: 'x-ms-summary'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1notifications~1{slug}~1{tracking_number}~1remove/post/parameters/0' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-summary' property is required. ValuePath: 'paths/~1notifications~1{slug}~1{tracking_number}~1remove/post/parameters/1' Value: 'x-ms-summary'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1notifications~1{slug}~1{tracking_number}~1remove/post/parameters/1' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-summary' property is required. ValuePath: 'paths/~1last_checkpoint~1{slug}~1{tracking_number}/get/parameters/0' Value: 'x-ms-summary'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1last_checkpoint~1{slug}~1{tracking_number}/get/parameters/0' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-summary' property is required. ValuePath: 'paths/~1last_checkpoint~1{slug}~1{tracking_number}/get/parameters/1' Value: 'x-ms-summary'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'description' property is required. ValuePath: 'paths/~1last_checkpoint~1{slug}~1{tracking_number}/get/parameters/1' Value: 'description'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error RequiredPropertyMissing: The 'x-ms-connector-metadata' property is required. ValuePath: '' Value: 'x-ms-connector-metadata'
##[error]/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json(,): error : Swagger file '/home/vsts/work/1/bbce340c-3edf-4c9b-93cc-e2295979aaed-files/apiDefinition.swagger.json' contains error(s).`

**Comment by taiki-yoshida** [comment]
@sriyen-msft I have fixed the validation errors. Kindly check the latest commit https://github.com/microsoft/PowerPlatformConnectors/pull/1260/commits/f4bc3249a27c33495545411ccd0dec4b16cbd850

**Comment by taiki-yoshida** [comment]
@sriyen-msft Kindly let me know if the latest commit fixed the issue. Thanks! https://github.com/microsoft/PowerPlatformConnectors/commit/f4bc3249a27c33495545411ccd0dec4b16cbd850

**Comment by sriyen-msft** [comment]
[[certify-connector]]

File: independent-publisher-connectors/AfterShip/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,2744 @@
...
+      "propertyName": "Website",
+      "propertyValue": "https://www.aftership.com"
+    },
+    {
+      "propertyName": "Privacy policy",
+      "propertyValue": "https://www.automizely.com/legal/privacy"
+    },
+    {
+      "propertyName": "Categories",
+      "propertyValue": "Shipment;Logistics"
```
Comment (sriyen-msft) [inline independent-publisher-connectors/AfterShip/apiDefinition.swagger.json:2741]: @taiki-yoshida  The connector category is invalid. You can only use maximum of two categories from the predefined list: AI, Business Management, Business Intelligence, Collaboration, Commerce, Communication, Content and Files, Finance, Data, Human Resources, Internet of Things, IT Operations, Lifestyle and Entertainment, Marketing, Productivity, Sales and CRM, Security, Social Media, Website

Docs - [Add metatda](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#step-3b-product-or-end-service-metadata)

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: independent-publisher-connectors/AfterShip/apiDefinition.swagger.json (hunk 7 above)
Comment (taiki-yoshida) [reply independent-publisher-connectors/AfterShip/apiDefinition.swagger.json]: Fixed to commerce

**Comment by sriyen-msft** [comment]
@taiki-yoshida Update - Your connector is now in production deployment and reached to [Preview](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#region-deployment) region, and it would be completed and available in all the production regions by Thursday 16h December 2021. Please let us know if you have any questions, Thanks.

**Comment by sriyen-msft** [comment]
@taiki-yoshida Apologies for the delay in response. 

Congratulations! The deployment of your connector and documentation page is now complete, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program, Thanks.

**Review by sriyen-msft — APPROVED** [review]

**Pull request merged**